devtools/cli/src/test/java/io/quarkus/cli/image/CliImageMavenTest.java

```diff
@@ -0,0 +1,68 @@
+package io.quarkus.cli.image;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.cli.CliDriver;
+import io.quarkus.devtools.testing.RegistryClientTestHelper;
+import picocli.CommandLine;
+
+/**
+ * Similar to CliProjecMavenTest ..
+ */
+public class CliImageMavenTest {
+    static Path workspaceRoot = Paths.get(System.getProperty("user.dir")).toAbsolutePath()
+            .resolve("target/test-project/CliImageMavenTest");
+    Path project;
+
+    @BeforeAll
+    public static void setupTestRegistry() {
+        RegistryClientTestHelper.enableRegistryClientTestConfig();
+    }
+
+    @AfterAll
+    public static void cleanupTestRegistry() {
+        RegistryClientTestHelper.disableRegistryClientTestConfig();
+    }
+
+    @BeforeEach
+    public void setupTestDirectories() throws Exception {
+        CliDriver.deleteDir(workspaceRoot);
+        project = workspaceRoot.resolve("code-with-quarkus");
+    }
+
+    @Test
+    public void testUsage() throws Exception {
+        CliDriver.Result result = CliDriver.execute(workspaceRoot, "create", "app", "-e", "-B", "--verbose");
+        Assertions.assertEquals(CommandLine.ExitCode.OK, result.getExitCode(), "Expected OK return code." + result);
+
+        // 1 image --help
+        result = CliDriver.execute(project, "image", "--help");
+        Assertions.assertTrue(result.getStdout().contains("Commands:"), "Should list subcommands\n");
```

Comment:
These are effectively the same tests as in devtools/cli/src/test/java/io/quarkus/cli/CliHelpTest.java. Validation of help output should move there.

What should be here are invocations with dry-run set, so you can verify what parameters were set based on the command line arguments used.

You have a lot of "if you set this argument, these properties should be set", and you should probably verify one or two, especially if there is anywhere "tricky" that future you won't remember to be careful about.

There should be at least one in maven, and one in gradle, that verifies you have the right resulting maven/gradle command (that's what dryrun is for)

Reply:
Tests moved to `CliHelpTest` and `CliImageMavenTest` now uses dry run to check that options are well respected.